### PR TITLE
Use URL parsing for Mission ID upload

### PR DESCRIPTION
**Pull request by Flamanis**
Parse Mission ID as int or as a URL. Use parsed URL to get the id paramstring.

Current Regex requires a character after the id numbers. If no character is found, then even a valid string will fail.

### CODE REVIEW

**Comment by Luminoscity** [comment]
No need to revert my commit, you actually need that. It doesn't work for this link without it `https://steamcommunity.com/sharedfiles/filedetails/?id=1517753537/`
The slash gets sees as part of the id and then it fails the isOnlyDigits check.

**Review by Luminoscity — CHANGES_REQUESTED** [review]
Need to put that back to conform to Sam's prettier standards and to handle the common single slash case.

**Comment by Flamanis** [comment]
It works perfectly fine on my end. I use the standard URL class, it handles it perfectly.

**Comment by Flamanis** [comment]
After looking through the workshop. Steam does not serve a page with a slash after the id. The test is mostly a moot point. But, it's not really extra code to run after the user already failed. It will very likely never have a positive, but eh.

File: src/lib/controls/Input.svelte [hunk 1]
```diff
@@ -49,13 +51,13 @@
 
 	function handleValidity(value: any, showErrors: boolean = true): boolean {
 		const validity = validate(value);
-		if (required) {
-			input.setCustomValidity(typeof validity === 'string' ? validity : validity ? '' : 'Invalid value.');
-			input.reportValidity();
+		if (required || validity !== null) {
+			input.setCustomValidity(typeof validity === 'string' ? validity : validity ? '' : 'Invalid value');
```
Comment (Luminoscity) [inline src/lib/controls/Input.svelte:55]: Breaks some validated inputs, like "Variant of" and "Mission Pack" and many other inputs on mission edit page (allows submission of form when invalid data is present in the box but silently replaces the input with the last valid input)

**Review by Luminoscity — CHANGES_REQUESTED** [review]

**Pull request merged**